src/core/lib/channel/handshaker_registry.cc

```diff
@@ -28,74 +30,85 @@
 // grpc_handshaker_factory_list
 //
 
-typedef struct {
-  grpc_handshaker_factory** list;
-  size_t num_factories;
-} grpc_handshaker_factory_list;
-
-static void grpc_handshaker_factory_list_register(
-    grpc_handshaker_factory_list* list, bool at_start,
-    grpc_handshaker_factory* factory) {
-  list->list = static_cast<grpc_handshaker_factory**>(gpr_realloc(
-      list->list,
-      (list->num_factories + 1) * sizeof(grpc_handshaker_factory*)));
+namespace grpc_core {
+
+namespace {
+
+class HandshakerFactoryList {
+ public:
+  void Register(bool at_start, UniquePtr<HandshakerFactory> factory);
+  void AddHandshakers(const grpc_channel_args* args,
+                      grpc_pollset_set* interested_parties,
+                      HandshakeManager* handshake_mgr);
+
+ private:
+  LinkedList<UniquePtr<HandshakerFactory>> factories_;
```

Comment:
You could use InlinedVector here. It moves the unique ptrs so you should be all safe.

Reply:
There are cases where new handshakers are added to the front of the list, which are slightly more efficient when using a linked list.  But this will generally only be done at grpc initialization time, so efficiency probably doesn't matter much.

Reply:
I see. Thank you for the details @markdroth ! I forgot about `at_start`. 

Based on your explanation, given that this is initialized once and read many times, InlinedVector might still more appealing given, as it'd faster to iterate over. I have no strong preference though. It's your and Arjun's call.

Reply:
Actually, I think that's a good argument: it seems better to optimize for things that happen at connection-establishment time rather than things that happen only at grpc initialization time.

Arjun, let's go ahead and use `InlinedVector<>` here.  That way, there's no need for a linked list implementation.

Reply:
+1 Thank you @markdroth !

Reply:
Ack.

Reply:
I've added an iterator based insert to InlinedVector; will update this PR once the current test suites have had a chance to run to completion.